src/CamScripterAPI.ts

```diff
@@ -0,0 +1,222 @@
+import * as WebSocket from 'ws';
+import * as EventEmitter from 'events';
+
+import { Digest } from './Digest';
+
+export type CamScripterOptions = {
+    tls?: boolean;
+    tlsInsecure?: boolean; // Ignore HTTPS certificate validation (insecure)
+    ip?: string;
+    port?: number;
+    auth?: string;
+};
+
+export type Declaration = {
+    type?: '' | 'SOURCE' | 'DATA';
+    namespace: string;
+    key: string;
+    value: string | boolean | number;
+    value_type: 'STRING' | 'INT' | 'BOOL' | 'DOUBLE';
+    key_nice_name?: string;
+    value_nice_name?: string;
+};
+
+export type EventDeclaration = {
+    declaration_id: string;
+    stateless: boolean;
+    declaration: Declaration[];
+};
+
+export type EventUndeclaration = {
+    declaration_id: string;
+};
+
+export type EventData = {
+    namespace: string;
+    key: string;
+    value: string | boolean | number;
+    value_type: 'STRING' | 'INT' | 'BOOL' | 'DOUBLE';
+};
+
+export type Event = {
+    declaration_id: string;
+    event_data: EventData[];
+};
+
+export type Response = {
+    call_id: number;
+    message: string;
+};
+
+type Message = {
+    call_id: number;
+    command: string;
+    data: any;
+};
+
+type AsyncMessage = {
+    resolve;
+    reject;
+};
+
+export class CamScripterAPI extends EventEmitter {
```

Comment:
can we name this class like `CamScripterAPICameraEventsGenerator` ?

Reply:
The name is extremely long and it looks a little bit strange in the code but it is descriptive and probably right direction when we want to split CO API as well. Changed!